### PR TITLE
Fixes for issues #129, #135

### PR DESCRIPTION
**Pull request by sholly**
This pull request contains fixes for cime issues #129, #135.

### CODE REVIEW

File: utils/perl5lib/Batch/BatchMaker.pm [hunk 1]
```diff
@@ -549,7 +571,42 @@ sub setCESMRun()
...
+				{
+					my $attrName = $attr->getName();
+					my $attrValue = $attr->getValue();
+					next if($attrValue eq 'default');
+					my $lcAttrName = lc $attrName;
+					if(defined $self->{$lcAttrName} && (lc $self->{$attrName} eq lc $attrValue))
+					{
+						$attrMatch = 1;
+						next;
+					}
```
Comment (quantheory) [inline utils/perl5lib/Batch/BatchMaker.pm]: Can't we remove the block above? If we enter that block, `$attrMatch` will always be `1`, so we shouldn't need to change it. Also, if the conditional above succeeds, then the one below will fail and we will just move on to the next iteration, so we shouldn't need a `next`, unless I'm missing something.

**Comment by quantheory** [comment]
Have you tested this fix for #135 on yellowstone, and checked that threaded runs really are using `hybrid_launch` instead of `launch`? I believe that there's an additional issue where the `hybrid_launch` line in `config_machines.xml` is broken. Currently, the line is:

``` sh
MP_PE_AFFINITY= MP_TASK_AFFINITY= MP_CPU_BIND_LIST= mpirun.lsf $ENV{'CESMDATAROOT'}/tools/bin/hybrid_launch 
```

However, when I tried that, it seemed to be broken, because we actually want to guarantee that those variables are unset, not that they are set to the null string. I think the line should be one of these instead:

``` sh
unset MP_PE_AFFINITY; unset MP_TASK_AFFINITY; unset MP_CPU_BIND_LIST; mpirun.lsf $ENV{'CESMDATAROOT'}/tools/bin/hybrid_launch

# Or this may be better:
unset MP_PE_AFFINITY && unset MP_TASK_AFFINITY && unset MP_CPU_BIND_LIST && mpirun.lsf $ENV{'CESMDATAROOT'}/tools/bin/hybrid_launch
```

Aside from this line being wrong, and the weird (unnecessary?) block I already commented on, it looks to me like this will work.

**Comment by sholly** [comment]
Re #135, I did figure out that this does not work: 'MP_PE_AFFINITY= MP_TASK_AFFINITY= MP_CPU_BIND_LIST=' doesn't work.  I actually implemented your first suggestion just as you said: 
unset MP_PE_AFFINITY; unset MP_TASK_AFFINITY; unset MP_CPU_BIND_LIST;
and this works.  This will get easier when my new module implementation goes in, we'll be able to conditionally set environment variables based on the compiler, threaded vs non-threaded, etc.

**Comment by quantheory** [comment]
Oh yeah, you're right, that change is right here in this PR. I don't know why I didn't see it...

**Comment by sholly** [comment]
The spurious for loop has been removed...

**Pull request merged**